File: challenge101/challenge101.py
"""
To use the python script in Command Prompt:
Write 'python your_file_name.py'
"""

# Four different arrays
numbers = [5, 10, 50, 75, 250]
moreNumbers = [540, 41, 78, 91, 2]
onlyTwoNumbers = [0, 1]
onlyOneNumber = [5]

# Function that finds the minimum and maximum numbers from an array
def FindMinMax(arrayOfNumbers):
    # Uses the built-in function min() to find the lowest number
    minimumNumber = min(arrayOfNumbers)
    # Uses the built-in function max() to find the largest number
    maximumNumber = max(arrayOfNumbers)
    # Prints a message with a normal string
    print("From the list specified:")
    # Prints the minimum and maximum number using a formatted (f) string
    print(f"Minimum is {minimumNumber}")
    print(f"Maximum is {maximumNumber}\n")

# Calls the functions with arguments
FindMinMax(numbers)
FindMinMax(moreNumbers)
FindMinMax(onlyTwoNumbers)
FindMinMax(onlyOneNumber)

File: challenge101/challenge101.js
// To run this script in Command Prompt: 
// write 'node your_file_name.js'
// example: 'node challenge101.js'

// Four different arrays
const numbers = [5, 10, 50, 75, 250];
const moreNumbers = [540, 41, 78, 91, 2];
const onlyTwoNumbers = [0, 1];
const onlyOneNumber = [5];

// A function with a parameter, requiring an array
function FindMinMax (arrayOfNumbers){
    // Finds the min number with the built-in Math.min() function
    const minimumNumber = Math.min(...arrayOfNumbers);
    // Finds the max number with the built-in Math.max() function
    const maximumNumber = Math.max(...arrayOfNumbers);

    // Log a message with the values from the list
    // Logs both minimum and maximum number
    console.log("Find the minimum and maximum of this list.");
    console.log("Minimum is " + minimumNumber);
    console.log("Maximum is " + maximumNumber + '\n');
}

// Function calls
FindMinMax(numbers);
FindMinMax(moreNumbers);
FindMinMax(onlyTwoNumbers);
FindMinMax(onlyOneNumber);